guides/git-workflow.md

```diff
@@ -1,21 +1,89 @@
+
+### Branching Workflow <a name="branching-pros-cons"></a>
+
+**Pros:**
+- Continuous integration tools such as Travis or Jenkins can be easily enabled for all branches of the repository, so new feature branches can automatically be tested when a pull request is opened or modified.
+
+**Cons:**
+- It is not possible for external collaborators to contribute unless they explicitly have write access to the `spacetelescope` repository.
+- It is easier to accidentally delete or overwrite the work of others who are also working on the same branch, or happen to have their own branch of the same name.  While it is possible to recover this work or to restrict the editing of specific branches to only select developers, it can sometimes be a painstaking process.
+- Repositories may accrue a fair amount of stale branches over time, which may become burdensome for developers, and may cause collaborators to infer that the project is not being maintained.
+
+
+### Forking Workflow <a name="forking-pros-cons"></a>
+
+**Pros:**
+- Promotes "open development", wherein potential external collaborators who are interested in contributing may do so despite not having write access to the `spacetelescope` repository.
+- Provides a consistent experience and barrier-to-entry for everyone involved in the project, whether you are an internal or external collaborator.
+- Allows for an explicit namespace in pull requests and repository history that makes it more apparent who "owns" which branch.
+- Allows for contributors to experiment with various repository settings and/or continuous integrations tools without affecting how the `spacetelescope` repository operates.
+
+**Cons:**
+- Requires extra work to keep up-to-date with the parent, 'upstream' repository (i.e. one must periodically "update" their personal fork).
+- If forks are of a public repository, and more than one person wants to collaborate on someone's personal fork, then one must add each colleague to your fork.
+- If forks are of a private repository, then everyone watching the parent repository is automatically subscribed to all of the fork activity by default.  These notifications may become burdensome.
+
+## Workflow Instructions <a name="workflow-instructions"></a>
+
+Below we provide step-by-step instructions on how to employ each workflow.
+
+### Opening a GitHub Issue <a name="github-issue"></a>
+Regardless of which workflow you chose, before contributing, you should determine if the eventual change is significant enough to warrant a GitHub issue; if you think that this change will be solving a significant problem or providing a significant enhancement to the project, then it would be advantageous to open an issue under the `spacetelescope` repository. This will allow both contributors and repository maintainers to keep track of the project and capture any needed context of the particular change. Any appropriate individuals should be assigned to the issue, and any appropriate label(s) should be tagged.
```

Comment:
```suggestion
Regardless of which workflow you choose, before contributing, you should determine if the eventual change is significant enough to warrant a GitHub issue; if you think that this change will be solving a significant problem or providing a significant enhancement to the project, then it would be advantageous to open an issue under the `spacetelescope` repository. This will allow both contributors and repository maintainers to keep track of the project and capture any needed context of the particular change. Any appropriate individuals should be assigned to the issue, and any appropriate label(s) should be tagged.

Meanwhile, for trivial fixes (e.g., typo), a pull request without an accompanying issue would be acceptable.
```

Reply:
This is great, thanks!